src/main/scala/com/microsoft/hyperspace/index/dataskipping/util/DataSkippingFileIndex.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (2021) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.index.dataskipping.util
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.{Column, DataFrame, SparkSession}
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.execution.datasources._
+import org.apache.spark.sql.functions.isnull
+import org.apache.spark.sql.types.StructType
+
+import com.microsoft.hyperspace.index.{FileIdTracker, IndexConstants}
+
+class DataSkippingFileIndex(
+    sparkSession: SparkSession,
+    fileIdTracker: FileIdTracker,
+    indexData: DataFrame,
+    private[dataskipping] val indexDataPred: Expression, // exposed for test
+    private[dataskipping] val baseFileIndex: FileIndex)
+    extends FileIndex {
+
+  override def listFiles(
+      partitionFilters: Seq[Expression],
+      dataFilters: Seq[Expression]): Seq[PartitionDirectory] = {
+    val partitions = baseFileIndex.listFiles(partitionFilters, dataFilters)
```

Comment:
Is the list after partition pruning on spark side?

Also I wonder if partition sketch is still effective after Spark's partition pruning?

Reply:
`baseFileIndex.listFiles()` will do the pruning if possible.

Partition sketches can be still effective for filters like `P = 1 or Q = 2` and `P = Q` where P is a partitioned column and Q is an indexed column.